hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractCallHandler.java

```diff
@@ -55,7 +55,8 @@ public void handle(@NonNull final HandleContext context) throws HandleException
         // Assemble the appropriate top-level record for the result
         context.recordBuilder(ContractCallRecordBuilder.class)
                 .contractCallResult(outcome.result())
-                .contractID(outcome.recipientIdIfCalled());
+                .contractID(outcome.recipientIdIfCalled())
+                .withGasFee(outcome.gasCostInTinybar());
         throwIfUnsuccessful(outcome.status());
```

Comment:
This code here and then also in `ContractCreateHandler` L62-64 makes me wonder if a `ContractCallRecordBuilder` should also have a method that just takes a `CallOutcome` and sets these fields itself.

Not sure whether that would be more clear or less clear though.

Reply:
Ah yes that seems much nicer; will fix in next PR also.